src/protonuke/protonuke.go

```diff
@@ -15,34 +15,39 @@ import (
-	f_https       = flag.Bool("https", false, "enable https (TLS) service")
-	f_httproot    = flag.String("httproot", "", "serve directory with http(s) instead of the builtin page generator")
-	f_httpGzip    = flag.Bool("httpgzip", false, "gzip image served in http/https pages")
-	f_httpCookies = flag.Bool("httpcookies", false, "enable cookie jar in http/https clients")
-	f_ssh         = flag.Bool("ssh", false, "enable ssh service")
-	f_smtp        = flag.Bool("smtp", false, "enable smtp service")
-	f_smtpUser    = flag.String("smtpuser", "", "specify a particular user to send email to for the given domain, otherwise random")
-	f_smtpTls     = flag.Bool("smtptls", true, "enable or disable sending mail with TLS")
-	f_smtpmail    = flag.String("smtpmail", "", "send email from a given file instead of the builtin email corpus")
-	f_mean        = flag.Duration("u", time.Duration(1000*time.Millisecond), "mean time between actions")
-	f_stddev      = flag.Duration("s", time.Duration(0), "standard deviation between actions")
-	f_min         = flag.Duration("min", time.Duration(0), "minimum time allowable for events")
-	f_max         = flag.Duration("max", time.Duration(60000*time.Millisecond), "maximum time allowable for events")
-	f_v4          = flag.Bool("ipv4", true, "use IPv4. Can be used together with -ipv6")
-	f_v6          = flag.Bool("ipv6", true, "use IPv6. Can be used together with -ipv4")
-	f_report      = flag.Duration("report", time.Duration(10*time.Second), "time between reports, set to 0 to disable")
-	f_httpTLSCert = flag.String("httptlscert", "", "file containing public certificate for TLS")
-	f_httpTLSKey  = flag.String("httptlskey", "", "file containing private key for TLS")
-	f_tlsVersion  = flag.String("tlsversion", "", "Select a TLS version for the client: tls1.0, tls1.1, tls1.2")
+	f_serve           = flag.Bool("serve", false, "act as a server for enabled services")
+	f_dns             = flag.Bool("dns", false, "enable dns service")
+	f_dnsv4           = flag.Bool("dnsv4", false, "dns client only requests type A records")
+	f_dnsv6           = flag.Bool("dnsv6", false, "dns client only requests type AAAA records")
+	f_randomhosts     = flag.Bool("random-hosts", false, "if no host range is supplied return a randomly generated ip")
+	f_http            = flag.Bool("http", false, "enable http service")
+	f_https           = flag.Bool("https", false, "enable https (TLS) service")
+	f_httproot        = flag.String("httproot", "", "serve directory with http(s) instead of the builtin page generator")
+	f_httpGzip        = flag.Bool("httpgzip", false, "gzip image served in http/https pages")
+	f_httpCookies     = flag.Bool("httpcookies", false, "enable cookie jar in http/https clients")
+	f_ftp             = flag.Bool("ftp", false, "enable ftp service")
+	f_ftps            = flag.Bool("ftps", false, "enable ftp (TLS) service")
+	f_ftpport         = flag.Int("ftpport", 21, "port to use for ftp client and server")
```

Comment:
We don't have port flags for the other protocols, is there a reason to support ftp on a non-standard port?